velox/exec/RowContainer.h

```diff
@@ -255,9 +288,28 @@ class RowContainer {
     return rows_.allocatedBytes() + stringAllocator_.retainedSize();
   }
 
+  // Returns the number of fixed size rows that can be allocated
+  // without growing the container and the number of unused bytes of
+  // reserved storage for variable length data.
+  std::pair<uint64_t, uint64_t> freeSpace() {
```

Comment:
std::pair<uint64_t, uint64_t> is not descriptive and it will be confusing to tell what the values are. Consider, adding a special-purpose struct with descriptive field names.